index.js

```diff
@@ -301,10 +308,51 @@ function registerPlugin (fastify, meta, allPlugins, parentPlugins = {}) {
   meta.registered = true
 }
 
+/**
+ * Used to determine if the contents of a required autoloaded file matches
+ * the shape of a Fastify route configuration object.
+ *
+ * @param {*} input The data to check.
+ *
+ * @returns {boolean} True if the data represents a route configuration object.
+ * False otherwise.
+ */
+function isRouteObject (input) {
+  if (input &&
+    Object.prototype.toString.call(input) === '[object Object]' &&
+    Object.prototype.hasOwnProperty.call(input, 'method')) {
+    return true
+  }
+  return false
+}
+
+/**
+ * Used to determine if the contents of a required autoloaded file is a valid
+ * plugin or route configuration object. In the case of a route configuraton
+ * object, it will later be wrapped into a plugin.
+ *
+ * @param {*} input The data to check.
+ *
+ * @returns {boolean} True if the object can be used by the autoload system by
+ * eventually passing it into `avvio`. False otherwise.
+ */
+function isPluginOrModule (input) {
+  let result = false
+
+  const inputType = Object.prototype.toString.call(input)
+  if (/\[object (AsyncFunction|Function|Module)\]/.test(inputType) === true) {
```

Comment:
Just for my curiosity:

What about generators? I mean.. they are also some kind of function

Reply:
Can you provide an instance where a generator is a valid Fastify plugin function?